libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -122,6 +122,18 @@ bool ImmutableValidator::visit(WhileStatement const& _whileStatement)
 	return false;
 }
 
+void ImmutableValidator::endVisit(IdentifierPath const& _identifierPath)
+{
+	if (auto const callableDef = dynamic_cast<CallableDeclaration const*>(_identifierPath.annotation().referencedDeclaration))
+		visitCallableIfNew(
+			*_identifierPath.annotation().requiredLookup == VirtualLookup::Virtual ?
```

Comment:
Can you try adding an assertion that this is always static lookup?

Reply:
Ah sorry, it actually can with a modifier, right?

Reply:
```
	if (
		dynamic_cast<CallableDeclaration const*>(_identifierPath.annotation().referencedDeclaration) &&
		_identifierPath.path().size() == 1
	)
		_identifierPath.annotation().requiredLookup = VirtualLookup::Virtual;
	else
		_identifierPath.annotation().requiredLookup = VirtualLookup::Static;
```

This is the logic that sets it.. so Yeah I'd say so.